### PR TITLE
Tee all logs to user log directory

### LINKED ISSUE #316: Establish logging into a persistent file within user's cache (e.g.  ~/.cache/dandi-cli/logs on linux) for each invocation
Probably should be just an additional log handler dumping the logs into such directory with a filename like `{datetime}-{pid}.log`.   
This way we could possibly later instruct users to share the logs for debugging and in general would be helpful.  
- Since we do not have a notion of "local dandiset" directory (e.g. like `.git/` for git) which is not to be shared/uploaded - I think user-wide cache directory is a good choice
- We need to make sure that logs reflect the path on which `dandi-cli` is invoked and what was the original command (`sys.argv`).  That would allow to associate logs with a particular dandiset when needed.
- We should log at INFO level the path to the log at the end of running the command.
- I wonder if we should make that log more verbose (DEBUG?) and/or provide a separate verbosity level , probably not for now.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #316 

Unfortunately, if the program exits abnormally, the final message giving the location of the logfile won't be printed.  I'm not aware of a way around this.

### CODE REVIEW

**Review by yarikoptic — CHANGES_REQUESTED** [review]
minor wish: uniform formatting within the log, so initial entries should also follow the format

File: dandi/cli/command.py [hunk 1]
```diff
@@ -80,6 +87,25 @@ def main(log_level, pdb=False):
...
     set_logger_level(get_logger(), log_level)
+
+    logdir = appdirs.user_log_dir("dandi-cli", "dandi")
+    logfile = os.path.join(
+        logdir, "{:%Y%m%d%H%M%SZ}-{}.log".format(datetime.utcnow(), os.getpid())
+    )
+    os.makedirs(logdir, exist_ok=True)
+    with open(logfile, "w") as fp:
+        print("sys.argv =", sys.argv, file=fp)
+        print("os.getcwd() =", os.getcwd(), file=fp)
```
[the remark this answers is not in the document]
Comment (jwodder) [reply dandi/cli/command.py]: Done (kind of kludgily).

**Comment by jwodder** [comment]
@yarikoptic I've figured out a way to ensure that the log file path is always printed, even on error.

File: dandi/cli/command.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yarikoptic) [reply dandi/cli/command.py]: thanks!  I think there should indeed be a better way (see e.g. how we handle filtering out of progress indication log messages in datalad: https://github.com/datalad/datalad/blob/master/datalad/log.py#L262 ) but let's proceed.

**Pull request merged**